src/features/instrument-details/components/pending-limit-orders.tsx

```diff
@@ -0,0 +1,144 @@
+import { useTranslation } from 'react-i18next';
+import { useQuery } from '@tanstack/react-query';
+import { RotateCw } from 'lucide-react';
+import type { ColumnDef } from '@tanstack/react-table';
+
+import type { LimitOrder } from '@/client';
+import { DataTable } from '@/features/shared/components/ui/data-table';
+import { Badge } from '@/features/shared/components/ui/badge';
+import { Button } from '@/features/shared/components/ui/button';
+import { Skeleton } from '@/features/shared/components/ui/skeleton';
+import { ErrorMessage } from '@/features/shared/components/error-message';
+import { ordersLimitListOptions } from '@/client/@tanstack/react-query.gen';
+import { withCurrency } from '@/features/shared/utils/numbers';
+import { TableCell, TableRow } from '@/features/shared/components/ui/table';
+import { EmptyMessage } from '@/features/shared/components/empty-message';
+
+interface PendingLimitOrdersProps {
+  ticker: string;
+}
+
+const REFETCH_INTERVAL_MS = 5000;
+
+export function PendingLimitOrders({ ticker }: PendingLimitOrdersProps) {
+  const { t } = useTranslation();
+
+  const { data, isPending, isError, isSuccess, refetch } = useQuery({
+    ...ordersLimitListOptions({ query: { ticker } }),
+    refetchInterval: REFETCH_INTERVAL_MS,
+  });
+
+  if (isSuccess && data.length === 0) {
+    return <EmptyMessage message={t('orders.no_pending_limit_orders')} />;
```

Comment:
if data length is 0, the refresh button should still be available, when i make an order, I have no way of refreshing if i havent made any previous waiting orders